### PR TITLE
[NEW] Service account one-tap login

### PR DESCRIPTION
**Pull request by bhardwajaditya**
service-account-one-tap-login

This pull request needs to be reviewed after the callbacks one.

### CODE REVIEW

File: app/models/server/models/Users.js [hunk 1]
```diff
@@ -20,6 +20,7 @@ export class Users extends Base {
 		this.tryEnsureIndex({ type: 1 });
 		this.tryEnsureIndex({ 'visitorEmails.address': 1 });
 		this.tryEnsureIndex({ federation: 1 }, { sparse: true });
+		this.tryEnsureIndex({ 'u._id': 1 });
```
Comment (kb0304) [inline app/models/server/models/Users.js:24]: Good

File: app/service-accounts/client/views/serviceAccountSidebarLogin.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+});
+
+Template.serviceAccountSidebarLogin.events({
+	'click .js-login'(e) {
+		e.preventDefault();
+		let { username } = this;
+		if (Meteor.user().u) {
+			username = Meteor.user().u.username;
+		}
+		console.log(username);
```
Comment (kb0304) [inline app/service-accounts/client/views/serviceAccountSidebarLogin.js]: Remove the log statement. Isn't the lint test supposed to give error on log statements?

File: app/service-accounts/server/methods/getLoginToken.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import { Meteor } from 'meteor/meteor';
+import { Accounts } from 'meteor/accounts-base';
+
+import { Users } from '../../../models';
+
+Meteor.methods({
+	getLoginToken(username) {
```
Comment (kb0304) [inline app/service-accounts/server/methods/getLoginToken.js:7]: What about the case when the user corresponding to neither `Meteor.userId` and nor `username` has a `.u` property?

File: app/service-accounts/client/views/serviceAccountSidebarLogin.js (hunk 2 above)
Comment (bhardwajaditya) [reply app/service-accounts/client/views/serviceAccountSidebarLogin.js]: Yes, it should, I will update this.

File: app/service-accounts/server/methods/getLoginToken.js (hunk 3 above)
Comment (bhardwajaditya) [reply app/service-accounts/server/methods/getLoginToken.js:7]: Ohh yes, I need to throw an error for that.

Comment (kb0304) [reply app/service-accounts/server/methods/getLoginToken.js:7]: Due to high security concerns in this method, I'd recommend structuring it like..  for the case we want to return, generate and return.. for everything else raise Error

**Review by kb0304 — APPROVED** [review]

**Pull request merged**